diff --git a/src/WallpaperEngine/Render/Drivers/Output/WaylandOutputViewport.cpp b/src/WallpaperEngine/Render/Drivers/Output/WaylandOutputViewport.cpp
index 35a9bfa..6c6c743 100644
--- a/src/WallpaperEngine/Render/Drivers/Output/WaylandOutputViewport.cpp
+++ b/src/WallpaperEngine/Render/Drivers/Output/WaylandOutputViewport.cpp
@@ -5,6 +5,7 @@
 #define namespace _namespace
 #define static
 extern "C" {
+#include <wayland-version.h>
 #include "wlr-layer-shell-unstable-v1-protocol.h"
 #include "xdg-output-unstable-v1-protocol.h"
 #include "xdg-shell-protocol.h"
@@ -76,7 +77,7 @@ static void scale (void* data, wl_output* wl_output, int32_t scale) {
     }
 }
 
-static void name (void* data, wl_output* wl_output, const char* name) {
+[[maybe_unused]] static void name (void* data, wl_output* wl_output, const char* name) {
     const auto viewport = static_cast<WaylandOutputViewport*> (data);
 
     if (name) {
@@ -87,7 +88,7 @@ static void name (void* data, wl_output* wl_output, const char* name) {
     viewport->getDriver ()->getOutput ().reset ();
 }
 
-static void description (void* data, wl_output* wl_output, const char* description) {
+[[maybe_unused]] static void description (void* data, wl_output* wl_output, const char* description) {
     // ignored
 }
 
@@ -104,8 +105,13 @@ static void surfaceFrameCallback (void* data, struct wl_callback* cb, uint32_t t
 
 constexpr struct wl_callback_listener frameListener = { .done = surfaceFrameCallback };
 
+#if defined(WAYLAND_VERSION_NUMBER) && WAYLAND_VERSION_NUMBER >= 12000
 constexpr wl_output_listener outputListener
     = { .geometry = geometry, .mode = mode, .done = done, .scale = scale, .name = name, .description = description };
+#else
+constexpr wl_output_listener outputListener
+    = { .geometry = geometry, .mode = mode, .done = done, .scale = scale };
+#endif
 
 constexpr struct zwlr_layer_surface_v1_listener layerSurfaceListener = {
     .configure = handleLSConfigure,
