diff --git a/vite.config.ts b/vite.config.ts
index ebbb417..be065bf 100644
--- a/vite.config.ts
+++ b/vite.config.ts
@@ -13,7 +13,7 @@ export default defineConfig({
 		svelte(),
 		electron({
 			main: {
-				entry: "src/frontend/main/main.ts",
+				entry: ["src/frontend/main/main.ts", "src/frontend/main/services/steamworksWorker.ts"],
 				vite: {
 					build: {
 						outDir: "build/frontend/electron",
