diff --git a/src/WallpaperEngine/Render/Objects/Effects/CPass.cpp b/src/WallpaperEngine/Render/Objects/Effects/CPass.cpp
index e93c293..4efed0e 100644
--- a/src/WallpaperEngine/Render/Objects/Effects/CPass.cpp
+++ b/src/WallpaperEngine/Render/Objects/Effects/CPass.cpp
@@ -861,7 +861,10 @@ void CPass::setupUniforms () {
     this->addUniform ("g_UserAlpha", renderable.getUserAlpha ());
     this->addUniform ("g_Alpha", renderable.getAlpha ());
     this->addUniform ("g_Color", renderable.getColor ());
-    this->addUniform ("g_Color4", renderable.getColor4 ());
+    const glm::vec4 color4 = renderable.getColor4 ();
+    this->addUniform (
+	"g_Color4", glm::vec4 (color4.r, color4.g, color4.b, color4.a * renderable.getAlpha ())
+    );
     if (!this->m_uniforms.contains ("g_CompositeColor")) {
 	this->addUniform ("g_CompositeColor", renderable.getCompositeColor ());
     }
