diff --git a/src/WallpaperEngine/Render/CTexture.cpp b/src/WallpaperEngine/Render/CTexture.cpp
index 475c482..298f345 100644
--- a/src/WallpaperEngine/Render/CTexture.cpp
+++ b/src/WallpaperEngine/Render/CTexture.cpp
@@ -188,9 +188,8 @@ void CTexture::setupOpenGLParameters (const uint32_t textureID) const {
     } else {
 	glTexParameteri (GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
 	glTexParameteri (GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+	glTexParameterf (GL_TEXTURE_2D, GL_TEXTURE_MAX_ANISOTROPY, 8.0f);
     }
-
-    glTexParameterf (GL_TEXTURE_2D, GL_TEXTURE_MAX_ANISOTROPY, 8.0f);
 }
 
 GLuint CTexture::getTextureID (const uint32_t imageIndex) const {
