diff --git a/CMakeLists.txt b/CMakeLists.txt
index 11565a1..def68d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,6 +152,11 @@ if(BUILD_TESTING)
 endif()
 # try to enable wayland builds when possible
 pkg_check_modules(WAYLAND_SUPPORT wayland-cursor wayland-protocols egl wayland-egl)
+if(WAYLAND_SUPPORT_FOUND)
+	message(STATUS "Wayland support: ON")
+else()
+	message(STATUS "Wayland support: OFF (missing wayland dev libraries - building X11 only)")
+endif()
 
 if(WAYLAND_SUPPORT_FOUND)
     find_program(WaylandScanner NAMES wayland-scanner)
